### PR TITLE
test: add backend API route test coverage

### PR DESCRIPTION
**Pull request by leowla**
## Issue

The backend API routes had almost no test coverage.

## Solution

Happy path and error/edge cases covered for each API route.

This PR increases the statement code coverage for:
- general API routes from 61.18% to 87.6%
- navigation API routes from 20.17% to 84.1%.
- DAOs from 59.79% to 83.34%.

## Risk

Low. Only changes to test suite.

## Checklist

- [x] Acceptance criteria met
- [x] Unit tests written and passing
- [x] Integration tests written and passing

## Summary by CodeRabbit

* **Tests**
  * Expanded backend API test coverage across access, collections, dashboard, files, groups, images, navigation, notes, resources, scenarios, scenes, staff, and users.
  * Added shared test helpers and improved server startup handling for more reliable automated testing.

* **Chores**
  * Updated test discovery settings so Jest picks up test files consistently.

### CODE REVIEW

File: backend/src/routes/api/__tests__/accessApi.test.js [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+  afterEach(async () => {
+    await mongoose.connection.db.dropDatabase();
+  });
+
+  afterAll(async () => {
+    server.close(async () => {
+      await mongoose.disconnect();
+      await mongoServer.stop();
+    });
+  });
```
[the remark this answers is not in the document]
Comment (leowla) [reply backend/src/routes/api/__tests__/accessApi.test.js]: outdated

File: .all-contributorsrc [hunk 2]
```diff
@@ -23,6 +23,90 @@
...
+      "avatar_url": "https://avatars.githubusercontent.com/u/202360294?v=4",
+      "profile": "https://github.com/justinedagreat",
+      "contributions": ["code", "design"]
+    },
+    {
+      "login": "",
+      "name": "Mekaella Millabas",
+      "avatar_url": "",
+      "profile": "",
+      "contributions": ["design"]
```
[the remark this answers is not in the document]
Comment (leowla) [reply .all-contributorsrc:101]: outdated

File: backend/src/db/daos/accessDao.js [hunk 3]
```diff
@@ -4,118 +4,60 @@ import User from "../models/user.js";
...
-    return false;
-  } catch {
-    return false;
-  }
+export const hasAccess = async (scenarioId, uid) => {
+  const user = await User.findOne({ uid }, { email: 1 }).lean();
+  if (!user) return false;
+  const access = await getAccessList(scenarioId);
+  return access.accessList.includes(user.email);
 };
```
[the remark this answers is not in the document]
Comment (leowla) [reply backend/src/db/daos/accessDao.js:24]: this is from updating the branch. out of scope of this pr.

**Review by harbassan — APPROVED** [review]
full transparency i havent read through every line of the 2500, but from a scan of each file and running the tests, it looks good to me, and very clean 👍 thanks for updating the access tests as well.
